test/commandLine/commandLineTest.js

```diff
@@ -0,0 +1,662 @@
+
+    testApp.on('message', params => {
+      assert.equal(params.nodeEnv, 'production')
+      assert.equal(params.htmlValidator.enable, false)
+      assert.equal(params.htmlValidator.separateProcess, true)
+      done()
+    })
+  })
+
+  it('should change the app to put it into production mode, disable validator, and detach validator ("-prod, -r, detach-validator")', function (done) {
+    const testApp = fork(path.join(appDir, 'app.js'), ['-prod', '-r', 'detach-validator'], {'stdio': ['pipe', 'pipe', 'pipe', 'ipc']})
+
+    testApp.on('message', params => {
+      assert.equal(params.nodeEnv, 'production')
+      assert.equal(params.htmlValidator.enable, false)
+      assert.equal(params.htmlValidator.separateProcess, true)
+      done()
+    })
+  })
+
+  it('should change the app to put it into production mode, disable validator, and detach validator ("-prod, disable-validator, --background-validator")', function (done) {
+    const testApp = fork(path.join(appDir, 'app.js'), ['-prod', 'disable-validator', '--background-validator'], {'stdio': ['pipe', 'pipe', 'pipe', 'ipc']})
+
+    testApp.on('message', params => {
+      assert.equal(params.nodeEnv, 'production')
+      assert.equal(params.htmlValidator.enable, false)
+      assert.equal(params.htmlValidator.separateProcess, true)
+      done()
+    })
+  })
+
+  it('should change the app to put it into production mode, disable validator, and detach validator ("-prod, disable-validator, -b")', function (done) {
+    const testApp = fork(path.join(appDir, 'app.js'), ['-prod', 'disable-validator', '-b'], {'stdio': ['pipe', 'pipe', 'pipe', 'ipc']})
+
+    testApp.on('message', params => {
+      assert.equal(params.nodeEnv, 'production')
+      assert.equal(params.htmlValidator.enable, false)
+      assert.equal(params.htmlValidator.separateProcess, true)
+      done()
+    })
+  })
+
+  /* check one letter combinations */
+
+  it('should change the app to put it into dev mode, enable validator, and attach validator ("-dha")', function (done) {
+    const testApp = fork(path.join(appDir, 'app.js'), ['-dha'], {'stdio': ['pipe', 'pipe', 'pipe', 'ipc']})
+
+    testApp.on('message', params => {
+      assert.equal(params.nodeEnv, 'development')
+      assert.equal(params.htmlValidator.enable, true)
+      assert.equal(params.htmlValidator.separateProcess, false)
+      done()
+    })
+  })
+
+  it('should change the app to put it into production mode, disable validator, and detach validator ("-prb")', function (done) {
+    const testApp = fork(path.join(appDir, 'app.js'), ['-prb'], {'stdio': ['pipe', 'pipe', 'pipe', 'ipc']})
+
+    testApp.on('message', params => {
+      assert.equal(params.nodeEnv, 'production')
+      assert.equal(params.htmlValidator.enable, false)
+      assert.equal(params.htmlValidator.separateProcess, true)
+      done()
+    })
+  })
+
+  /* check mutually exclusive flags  */
```

Comment:
Add a test to this section that tests the limits of mutually exclusive flags like `-dpabhr`